lib/pure/strmisc.nim

```diff
@@ -85,6 +86,124 @@ proc rpartition*(s: string, sep: string): (string, string, string)
+
+  func parseFloatThousandSep*(str: openArray[char]; options: set[ParseFloatOptions];
+      sep = ','; decimalDot = '.'): float =
+    ## Convenience func for `parseFloat` which allows for thousand separators,
+    ## this is designed to parse floats as found in the wild formatted for humans.
+    ##
+    ## Fine grained flexibility and strictness is up to the user,
+    ## you can set the `options` using `ParseFloatOptions` enum.
+    ##
+    ## `parseFloatThousandSep` "prepares" `str` and then calls `parseFloat`,
+    ## consequently `parseFloatThousandSep` by design is slower than `parseFloat`.
+    ##
+    ## The following assumptions and requirements must be met:
+    ## - `sep` must not be `'-'`.
+    ## - `decimalDot` must not be `'-'` nor `' '`.
+    ## - `sep` and `decimalDot` must be different.
+    ## - `str` must be stripped of trailing and leading whitespaces.
+    ## - `pfTrailingDot` and `pfLeadingDot` must not be used together.
+    ##
+    ## See also:
+    ## * `parseFloat <strutils.html#parseFloat,string>`_
+    runnableExamples:
+      doAssert parseFloatThousandSep("10,000.000", {}) == 10000.0
+      doAssert parseFloatThousandSep("1,000,000.000", {}) == 1000000.0
+      doAssert parseFloatThousandSep("10,000,000.000", {}) == 10000000.0
+      doAssert parseFloatThousandSep("1,222.0001", {}) == 1222.0001
+      doAssert parseFloatThousandSep("10.000,0", {}, '.', ',') == 10000.0
+      doAssert parseFloatThousandSep("1'000'000,000", {}, '\'', ',') == 1000000.0
+      doAssert parseFloatThousandSep("1000000", {pfDotOptional}) == 1000000.0
+      doAssert parseFloatThousandSep("-1,000", {pfDotOptional}) == -1000.0
+      ## You can omit `sep`, but then all subsequent `sep` to the left must also be omitted:
+      doAssert parseFloatThousandSep("1000,000", {pfDotOptional}) == 1000000.0
+      ## Examples using different ParseFloatOptions:
+      doAssert parseFloatThousandSep(".1", {pfLeadingDot}) == 0.1
+      doAssert parseFloatThousandSep("1", {pfDotOptional}) == 1.0
+      doAssert parseFloatThousandSep("1.", {pfTrailingDot}) == 1.0
+      doAssert parseFloatThousandSep("1.0,0,0", {pfSepAnywhere}) == 1.0
+      doAssert parseFloatThousandSep("", {pfEmptyString}) == 0.0
+
+    assert sep != '-' and decimalDot notin {'-', ' '} and sep != decimalDot
+    assert not(pfTrailingDot in options and pfLeadingDot in options)
+
+    proc parseFloatThousandSepRaise(i: int; c: char; s: openArray[char]) {.noinline, noreturn.} =
+      raise newException(ValueError,
+        "Invalid float containing thousand separators, invalid char $1 at index $2 for input $3" %
+        [c.repr, $i, s.repr])
+
+    # Fail fast, before looping.
+    let strLen = str.len
+    if strLen == 0: # Empty string.
+      if pfEmptyString notin options:
+        parseFloatThousandSepRaise(0, ' ', "empty string")
+      else:
+        return 0.0 # So user dont need to do `(try: parseFloat(str) except: 0.0)` etc
```

Comment:
sounds too niche and trivial to work without this flag in the rare case where needed, eg:
`let s = if str.len == 0: 0.0 else: parseFloatThousandSep(str)`